controlplane/pkg/prefix_pool/prefixpool.go

```diff
@@ -201,13 +205,44 @@ func ExtractPrefixes(prefixes []string, requests ...*connectioncontext.ExtraPref
 	if len(result) == 0 {
 		return nil, prefixes, fmt.Errorf("Failed to extract prefixes, there is no available %v", prefixes)
 	}
+
+	/* Verify the allocated addresses does not intersect with one from the excludedPrefix list */
+	err = verifyExcluding(result, excludedPrefixes)
+	if err != nil {
+		return nil, nil, err
+	}
+
 	return result, newPrefixes, nil
 }
 
+/* Check for excludedPrefixes intersection */
+func verifyExcluding(prefixes []string, excludedPrefixes []string) error {
+	_, ipNet, err := net.ParseCIDR(prefixes[0])
+	if err != nil {
+		return err
+	}
+
+	excludedPrefixesPool, err := NewPrefixPool(excludedPrefixes...)
+	if err != nil {
+		return err
+	}
+
+	intersect, err := excludedPrefixesPool.Intersect(ipNet.String())
+	if err != nil {
+		return err
+	}
+
+	if intersect {
+		return fmt.Errorf("IPAM: Allocated IP intersects with excludedPrefix")
+	}
+
+	return nil
+}
+
 func ExtractPrefix(prefixes []string, prefixLen uint32) (string, []string, error) {
```

Comment:
This is function where choosing of appropriate prefix happen. So we need to pass excludedPrefixes here, to affect on this algorithm.

There is a cycle over all available prefixes. Firstly it tries to find prefix with length equal to `prefixLen`. In case of absent of such prefix we find a candidate to split in 2 subnet. So please provide additional logic of choosing that candidate - it mustn't be equal to one of excludedPrefix or be included in one of excludedPrefix